Identity_Presentation/Controllers/AuthenticationController.cs

```diff
@@ -0,0 +1,38 @@
+﻿using Identity_Application.Commands.Authentication;
+using Identity_Application.Models.AuthorizationModels;
+using Identity_Application.Queries.Authentication;
+using MediatR;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+
+namespace Identity_Presentation.Controllers;
+
+[ApiController]
+[Route("api/[controller]")]
+public class AuthenticationController : Controller
+{
+    private readonly IMediator _mediator;
+
+    public AuthenticationController(IMediator mediator)
+    {
+        _mediator = mediator;
+    }
+
+    [AllowAnonymous]
+    [HttpPost("register")]
+    public async Task<IActionResult> Register([FromBody] RegisterVM vm)
+    {
+        var token = await _mediator.Send(new RegisterCommand(vm));
+
+        return Ok(token);
```

Comment:
what kind of token we return, do we do login in registration scope?

Reply:
Yes, as it was said previously, the registration is adding new user and logging in with the same data at the same time.

Reply:
I'm worried a bit about - Single responsibility in this case

Reply:
@Nazaziik